renku/ui/api/models/run.py

```diff
@@ -95,66 +95,233 @@ def __fspath__(self):
         return str(self._path)
 
     @property
-    def path(self):
+    def path(self) -> Path:
         """Return path of file."""
         return self._path
 
 
-class Input(_PathBase):
+class Parameter:
+    """API Parameter model."""
+
+    @ensure_project_context
+    def __init__(
+        self,
+        name: str,
+        value: Union[Path, str, bool, int, float],
+        project,
+        default_value: Union[Path, str, bool, int, float] = None,
+        description: str = None,
+        position: int = None,
+        prefix: str = None,
```

Comment:
I think we don't want to allow users setting this, and we should only set it in the class method below. Setting e.g. position in my user script wouldn't do anything, so it'd be a useless option for a user to set.

Description might be nice but we probably don't want to change the indirect parameter logic in this PR.

I'm a bit torn here, letting a user set something that does nothing isn't nice, but if we let users use the API to construct Plans in the future, it might be useful again. It'd be nice if Python allowed constructor overloading...